Goal: terminate argument validation in login when it hits a '--'.
Fixes: #66368

Status wrt upstream: It could certainly be submitted to upstream.
                     Upstream comment: "Better will be rewrite login
                     for use getopt_long()."

Index: shadow-4.0.14/src/login.c
===================================================================
--- shadow-4.0.14.orig/src/login.c	2006-01-03 08:23:48.172522068 +0100
+++ shadow-4.0.14/src/login.c	2006-01-03 08:25:01.557624818 +0100
@@ -252,6 +252,8 @@
 	for (arg = 1; arg < argc; arg++) {
 		if (argv[arg][0] == '-' && strlen (argv[arg]) > 2)
 			usage ();
+		if (!strcmp(argv[arg], "--"))
+			break; /* stop checking on a "--" */
 	}
 }
 
